drivers/sensor/sbs_gauge/sbs_gauge.c

```diff
@@ -0,0 +1,263 @@
+		break;
+
+	case SENSOR_CHAN_GAUGE_REMAINING_CHARGE_CAPACITY:
+		val->val1 = data->remaining_charge_capacity;
+		break;
+
+	case SENSOR_CHAN_GAUGE_NOM_AVAIL_CAPACITY:
+		val->val1 = data->nom_avail_capacity;
+		break;
+
+	case SENSOR_CHAN_GAUGE_FULL_AVAIL_CAPACITY:
+		val->val1 = data->full_avail_capacity;
+		break;
+
+	case SENSOR_CHAN_GAUGE_TIME_TO_EMPTY:
+		val->val1 = data->time_to_empty;
+		break;
+
+	case SENSOR_CHAN_GAUGE_TIME_TO_FULL:
+		val->val1 = data->time_to_full;
+		break;
+
+	case SENSOR_CHAN_GAUGE_CYCLE_COUNT:
+		val->val1 = data->cycle_count;
+		break;
+
+	default:
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+
+static int sbs_gauge_sample_fetch(const struct device *dev, enum sensor_channel chan)
+{
+	struct sbs_gauge_data *const data = (struct sbs_gauge_data *const)dev->data;
+	int status = 0;
+
+	switch (chan) {
+	case SENSOR_CHAN_GAUGE_VOLTAGE:
+		status = sbs_cmd_reg_read(dev, SBS_GAUGE_CMD_VOLTAGE, &data->voltage);
+		if (status < 0) {
+			LOG_ERR("Failed to read voltage");
+			return -EIO;
+		}
+		break;
+
+	case SENSOR_CHAN_GAUGE_AVG_CURRENT:
+		status = sbs_cmd_reg_read(dev, SBS_GAUGE_CMD_AVG_CURRENT, &data->avg_current);
+		if (status < 0) {
+			LOG_ERR("Failed to read average current ");
+			return -EIO;
+		}
+		break;
+
+	case SENSOR_CHAN_GAUGE_TEMP:
+		status = sbs_cmd_reg_read(dev, SBS_GAUGE_CMD_TEMP, &data->internal_temperature);
+		if (status < 0) {
+			LOG_ERR("Failed to read internal temperature");
+			return -EIO;
+		}
+		break;
+
+	case SENSOR_CHAN_GAUGE_STATE_OF_CHARGE:
+		status = sbs_cmd_reg_read(dev, SBS_GAUGE_CMD_ASOC, &data->state_of_charge);
+		if (status < 0) {
+			LOG_ERR("Failed to read state of charge");
+			return -EIO;
+		}
+		break;
+
+	case SENSOR_CHAN_GAUGE_FULL_CHARGE_CAPACITY:
+		status = sbs_cmd_reg_read(dev,
+			SBS_GAUGE_CMD_FULL_CAPACITY, &data->full_charge_capacity);
+		if (status < 0) {
+			LOG_ERR("Failed to read full charge capacity");
+			return -EIO;
+		}
+		break;
+
+	case SENSOR_CHAN_GAUGE_REMAINING_CHARGE_CAPACITY:
+		status = sbs_cmd_reg_read(dev,
+			SBS_GAUGE_CMD_REM_CAPACITY, &data->remaining_charge_capacity);
+		if (status < 0) {
+			LOG_ERR("Failed to read remaining charge capacity");
+			return -EIO;
+		}
+		break;
+
+	case SENSOR_CHAN_GAUGE_NOM_AVAIL_CAPACITY:
+		status = sbs_cmd_reg_read(dev,
+			SBS_GAUGE_CMD_NOM_CAPACITY, &data->nom_avail_capacity);
+		if (status < 0) {
+			LOG_ERR("Failed to read nominal available capacity");
+			return -EIO;
+		}
+		break;
+
+	case SENSOR_CHAN_GAUGE_FULL_AVAIL_CAPACITY:
+		status = sbs_cmd_reg_read(dev,
+			SBS_GAUGE_CMD_FULL_CAPACITY, &data->full_avail_capacity);
+		if (status < 0) {
+			LOG_ERR("Failed to read full available capacity");
+			return -EIO;
+		}
+		break;
+
+	case SENSOR_CHAN_GAUGE_TIME_TO_EMPTY:
+		status = sbs_cmd_reg_read(dev, SBS_GAUGE_CMD_AVG_TIME2EMPTY, &data->time_to_empty);
+
+		data->time_to_empty = (data->time_to_empty) & 0x00FF;
+
+		if (status < 0) {
+			LOG_ERR("Failed to read state of health");
```

Comment:
This same log message is being used a few times. Copy/paste error?

Reply:
good catch, that was a copy/paste error